libbeat/common/kubernetes/watcher.go

```diff
@@ -173,6 +181,18 @@ func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOption
 		}
 
 		objType = "statefulset"
+	case *Service:
+		svc := client.CoreV1().Services(opts.Namespace)
+		listwatch = &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return svc.List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return svc.Watch(options)
+			},
+		}
+
+		objType = "service"
```

Comment:
should the watcher refuse to start if `scope != cluster`? We should at least explain this in the docs if not

Reply:
updated the docs for the same. the watcher needs some refactor so that we can address some of these concerns and also allow it to do additional enrichment